openpype/hosts/max/plugins/publish/extract_tycache.py

```diff
@@ -0,0 +1,183 @@
+import os
+
+import pyblish.api
+from pymxs import runtime as rt
+
+from openpype.hosts.max.api import maintained_selection
+from openpype.pipeline import publish
+
+
+class ExtractTyCache(publish.Extractor):
+    """
+    Extract tycache format with tyFlow operators.
+    Notes:
+        - TyCache only works for TyFlow Pro Plugin.
+
+    Methods:
+        self.get_export_particles_job_args(): sets up all job arguments
+            for attributes to be exported in MAXscript
+
+        self.get_operators(): get the export_particle operator
+
+        self.get_files(): get the files with tyFlow naming convention
+            before publishing
+    """
+
+    order = pyblish.api.ExtractorOrder - 0.2
+    label = "Extract TyCache"
+    hosts = ["max"]
+    families = ["tycache"]
+
+    def process(self, instance):
+        # TODO: let user decide the param
+        start = int(instance.context.data["frameStart"])
+        end = int(instance.context.data.get("frameEnd"))
+        self.log.info("Extracting Tycache...")
+
+        stagingdir = self.staging_dir(instance)
+        filename = "{name}.tyc".format(**instance.data)
+        path = os.path.join(stagingdir, filename)
+        filenames = self.get_files(instance, start, end)
+        additional_attributes = instance.data.get("tyc_attrs", {})
+
+        with maintained_selection():
+            job_args = self.get_export_particles_job_args(
+                instance.data["members"],
+                start, end, path,
+                additional_attributes)
+            for job in job_args:
+                rt.Execute(job)
+        representations = instance.data.setdefault("representations", [])
+        representation = {
+            'name': 'tyc',
+            'ext': 'tyc',
+            'files': filenames if len(filenames) > 1 else filenames[0],
+            "stagingDir": stagingdir,
+        }
+        representations.append(representation)
+        self.log.info(f"Extracted instance '{instance.name}' to: {filenames}")
+
+        # Get the tyMesh filename for extraction
+        mesh_filename = f"{instance.name}__tyMesh.tyc"
+        mesh_repres = {
+            'name': 'tyMesh',
+            'ext': 'tyc',
+            'files': mesh_filename,
+            "stagingDir": stagingdir,
+            "outputName": '__tyMesh'
+        }
+        representations.append(mesh_repres)
+        self.log.info(
+            f"Extracted instance '{instance.name}' to: {mesh_filename}")
```

Comment:
```suggestion
        self.log.debug(f"Extracted instance '{instance.name}' to: {filenames}")

        # Get the tyMesh filename for extraction
        mesh_filename = f"{instance.name}__tyMesh.tyc"
        mesh_repres = {
            'name': 'tyMesh',
            'ext': 'tyc',
            'files': mesh_filename,
            "stagingDir": stagingdir,
            "outputName": '__tyMesh'
        }
        representations.append(mesh_repres)
        self.log.debug(
            f"Extracted instance '{instance.name}' to: {mesh_filename}")
```
Also, are you sure you want to report both with the same `Extracted instance ` message? Maybe a bit confusing to see twice?

Reply:
Maybe just do it once for tyCache part as we usually load tyCache instead of tyMesh